src/main/java/org/jboss/tools/intellij/componentanalysis/yarn/YarnCAAnnotator.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package org.jboss.tools.intellij.componentanalysis.yarn;
+
+import com.intellij.json.psi.JsonArray;
+import com.intellij.json.psi.JsonObject;
+import com.intellij.json.psi.JsonProperty;
+import com.intellij.json.psi.JsonStringLiteral;
+import com.intellij.json.psi.JsonValue;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.redhat.exhort.api.v4.DependencyReport;
+import org.jboss.tools.intellij.componentanalysis.CAAnnotator;
+import org.jboss.tools.intellij.componentanalysis.CAIntentionAction;
+import org.jboss.tools.intellij.componentanalysis.CAUpdateManifestIntentionAction;
+import org.jboss.tools.intellij.componentanalysis.Dependency;
+import org.jboss.tools.intellij.componentanalysis.VulnerabilitySource;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class YarnCAAnnotator extends CAAnnotator {
+
+    @Override
+    protected String getInspectionShortName() {
+        return YarnCAInspection.SHORT_NAME;
+    }
+
+    @Override
+    protected Map<Dependency, List<PsiElement>> getDependencies(PsiFile file) {
```

Comment:
This method seems a bit hard to follow. What about creating helper methods to make it more readable and maintainable?
Some suggestions:
```
private Set<String> getIgnoredDependencies(PsiFile file) {
 ...
}

private Map<Dependency, List<PsiElement> collectDependencies(PsiFile file, Set<String> ignored) {...}
```

Reply:
Fixed. Moved common codes to `CAUtil` and refactor methods by functionality.